data-extraction/data-import/src/main/java/com/wipro/ats/bdre/dataimport/HDFSImport.java

```diff
@@ -124,32 +124,28 @@ public int run(String[] param) throws Exception {
 
 
             //Checking the condition for importing by table ,columns or query
-            String importType = commonProperties.getProperty("import");
-            if (null != commonProperties.getProperty("query") && "" != commonProperties.getProperty("query")) {
+            if (null != commonProperties.getProperty(QUERY_STRING) && "" != commonProperties.getProperty(QUERY_STRING)) {
                 options.setSqlQuery(query);//import using the query
 
             } else {
-                if (null != commonProperties.getProperty("columns")) {
-                    if (size != 0) {
+                if (null != commonProperties.getProperty("columns") && size != 0) {
 
                         options.setTableName(tableName);
                         options.setColumns(cols);        //importing table or columns
 
-                    }
                 }
 
                 if (!("None".equalsIgnoreCase(incrementMode)) && incrementMode != null) {
                     ProcessLog processLog = new ProcessLog();
                     ProcessLogInfo processLogInfo = new ProcessLogInfo();
                     String logCategory;
-                    logCategory = "IncrementalImport";
+                    logCategory = LOG_CATEGORY;
                     String msgId = "last value";
                     processLogInfo = processLog.getLastValue(processId, msgId, logCategory);
                     if (processLogInfo != null) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS134)